src/ws/session-utils.c

```diff
@@ -47,6 +48,93 @@ static size_t auth_msg_size = 0;
 static FILE *authf = NULL;
 
 
+static bool
+char_needs_json_escape (char c)
+{
+  return c < ' ' || c == '\\' || c == '"';
+}
+
+static bool
+json_escape_char (FILE *stream,
+                  char c)
+{
+  if (c == '\\')
+    return fputs ("\\\\", stream) >= 0;
+  else if (c == '"')
+    return fputs ("\\\"", stream) >= 0;
+  else
+    return fprintf (stream, "\\u%04x", c) == 6;
+}
+
+static bool
+json_escape_string (FILE       *stream,
+                    const char *str,
+                    size_t      maxlen)
+{
+  size_t offset = 0;
+
+  while (offset < maxlen && str[offset])
+    {
+      size_t start = offset;
+
+      while (offset < maxlen && str[offset] && !char_needs_json_escape (str[offset]))
+        offset++;
+
+      fprintf (stderr, "off%zu st%zu\n", offset, start);
+
+      /* print the non-escaped prefix, if there is one */
+      if (offset != start)
+        {
+          size_t length = offset - start;
+          if (fwrite (str + start, 1, length, stream) != length)
+            return false;
+        }
+
+      /* print the escaped character, if there is one */
+      if (offset < maxlen && str[offset])
+        {
+          if (!json_escape_char (stream, str[offset]))
+            return false;
+
+          offset++;
+        }
+    }
+
+  return true;
+}
+
+bool
+json_print_string_property (FILE       *stream,
+                            const char *key,
+                            const char *value,
+                            ssize_t     maxlen)
+{
+  size_t expected = strlen (key)  + 8;
```

Comment:
When I count the chars in the string below, I keep getting 7 (ignoring the %s which is the strlen(key)). But the terminating 0 byte is not counted usually. I suppose this works, but it would be nice to add a comment that explains the 8.

Also, please drop the duplicated space.

Reply:
This is actually a good catch.  The incorrect number here was being matched by the incorrectly inverted `==` vs `!=` below.  I had had 7 here but it wasn't working before, and I recounted, (probably accidentally including the `%s`) and got 8, changed it to that, and of course that caused it to work.... Oops.